### ISSUE TITLE
Lifted `vmap` in linen does not split rng_key when single key is passed to `apply`

### ISSUE THREAD

**Issue by krzysztofrusek** [google/flax#4973]
Passing single rig key to vaped module `apply` where `vmap` is applied only across streams, does not split key correctly.

### System information
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Darwin Kernel Version 24.6.0
- Flax, jax, jaxlib versions (obtain with `pip show flax jax jaxlib`:
```
 uv pip show flax jax jaxlib
Name: flax
Version: 0.10.2
Location: /Users/krzysiek/Documents/ml4wifi/ltc/.venv/lib/python3.12/site-packages
Requires: jax, msgpack, numpy, optax, orbax-checkpoint, pyyaml, rich, tensorstore, typing-extensions
Required-by: jax-ai-stack
---
Name: jax
Version: 0.5.0
Location: /Users/krzysiek/Documents/ml4wifi/ltc/.venv/lib/python3.12/site-packages
Requires: jaxlib, ml-dtypes, numpy, opt-einsum, scipy
Required-by: chex, flax, jax-ai-stack, optax, orbax-checkpoint, orbax-export
---
Name: jaxlib
Version: 0.5.0
Location: /Users/krzysiek/Documents/ml4wifi/ltc/.venv/lib/python3.12/site-packages
Requires: ml-dtypes, numpy, scipy
Required-by: chex, jax, optax, orbax-export
```
- Python version: 3.12
- GPU/TPU model and memory:
- CUDA version (if applicable):

### Steps to reproduce:

Minimal example

```python
import unittest
import unittest

import jax
import jax.numpy as jnp


import flax.linen as nn

class MyTestCase(unittest.TestCase):
    def test_something(self):

        x = jnp.ones((5, 4))

        Batched = nn.vmap(
            nn.Sequential,
            in_axes=(None,), out_axes=-1,
            variable_axes={'params': None},
            split_rngs={'params': False, 'dropout': True},
            axis_size=10
        )
        mod = Batched([
            nn.Dense(2),
            nn.Dropout(0.1,deterministic=False)])

        vars = mod.init(jax.random.key(0), x)
        y = mod.apply(vars, x, rngs=jax.random.key(0))
        s=jnp.std(y, axis=-1)
        self.assertTrue(jnp.all(jnp.allclose(s, 0)))

        y = mod.apply(vars, x, rngs=dict(params=jax.random.key(0), dropout=jax.random.key(1)))
        s=jnp.std(y, axis=-1)
        self.assertFalse(jnp.all(jnp.allclose(s, 0)))

        ...



if __name__ == '__main__':
    unittest.main()

```

**Comment by vfdev-5** [comment]
@krzysztofrusek thansk for the report. Running your code with the latest flax and jax ('0.7.2', '0.12.0'), it is matter of precision that the line `self.assertTrue(jnp.all(jnp.allclose(s, 0)))` is failing. If you print the value of s it is 
```
[[5.9604645e-08 5.9604645e-08]
 [5.9604645e-08 0.0000000e+00]
 [5.9604645e-08 5.9604645e-08]
 [5.9604645e-08 5.9604645e-08]
 [0.0000000e+00 0.0000000e+00]]
```
and also if you set `axis_size` to a smaller value, e.g. 5, assert will pass.

**Comment by krzysztofrusek** [comment]
I checked in the lates jax/flax and still in my vent there is the same effect.
Maybe assert on `std` is not the best way to check I the variability across ensemble is not randomized.
You can inspect `y[2,1,:]`, and they are exactly the same numbers.

**Comment by vfdev-5** [comment]
> You can inspect y[2,1,:], and they are exactly the same numbers.

@krzysztofrusek I was thinking that you were expecting that in your test. Can you state more explicitly what do you see and what do you expect to see and why? Thanks!

**Comment by krzysztofrusek** [comment]
> > You can inspect y[2,1,:], and they are exactly the same numbers.
> 
> [@krzysztofrusek](https://github.com/krzysztofrusek) I was thinking that you were expecting that in your test. Can you state more explicitly what do you see and what do you expect to see and why? Thanks!

> what do you see 

When I run `mod.apply(vars, x, rngs=jax.random.key(0))` all the values across ensemble are identical

> and what do you expect to see and why?

I expect them to be different because I specified dropout rings to be sales: `split_rngs={'params': False, 'dropout': True}` and this is what I get in this call `mod.apply(vars, x, rngs=dict(params=jax.random.key(0), dropout=jax.random.key(0)))`

So the better test would be this one (not the that I use exactly the same key so I can expect the same results.

```python
import unittest

import jax
import jax.numpy as jnp

import flax.linen as nn

class MyTestCase(unittest.TestCase):
    def test_something(self):

        x = jnp.ones((6, 4))

        Batched = nn.vmap(
            nn.Sequential,
            in_axes=(None,), out_axes=-1,
            variable_axes={'params': None},
            split_rngs={'params': False, 'dropout': True},
            axis_size=100
        )
        mod = Batched([
            nn.Dense(2),
            nn.Dropout(0.1,deterministic=False)])

        vars = mod.init(jax.random.key(0), x)
        y1 = mod.apply(vars, x, rngs=jax.random.key(0))
        # s=jnp.std(y, axis=-1)
        # self.assertFalse(jnp.all(jnp.allclose(s, 0)))

        y2 = mod.apply(vars, x, rngs=dict(params=jax.random.key(0), dropout=jax.random.key(0)))
        # s=jnp.std(y, axis=-1)
        # self.assertFalse(jnp.all(jnp.allclose(s, 0)))
        self.assertTrue(jnp.allclose(y1, y2))
        print(y1[0,0,:])
        print(y2[0, 0, :])

        ...



if __name__ == '__main__':
    unittest.main()

```

**Comment by vfdev-5** [comment]
I tend to say that the result of `y1 = mod.apply(vars, x, rngs=jax.random.key(0))` is expected.
When provided rngs as a non-dict, it will be transformed into `rngs={"params": rngs}`
See https://github.com/google/flax/blob/6080725f757a5092d13fb43210ba0abc362823cc/flax/linen/module.py#L2215-L2221
Next, when we create random variables or inside Dropout, there is a call of `Module.make_rng()` which internally calls `score.make_rng` which has a fallback to "params":
https://github.com/google/flax/blob/6080725f757a5092d13fb43210ba0abc362823cc/flax/core/scope.py#L724-L728

In this case, whatever the value of "dropout" in `split_rngs={'params': False, 'dropout': True}`, split logic will be dicted by "params" key as the only rngs provided in apply call.

Similar logic applies to `Module.init` and variables creation:
```python
import jax
import jax.numpy as jnp
import flax.linen as nn

class Model(nn.Module):
  @nn.compact
  def __call__(self, x):
    x = nn.Dense(2)(x)
    kernel = self.variable(
      'other_collection',
      'kernel',
      lambda: jax.random.normal(self.make_rng('other'), x.shape),
    )

    return x + kernel.value

BatchModel = nn.vmap(
  Model,
  in_axes=0,
  out_axes=0,
  variable_axes={'params': 0, 'other_collection': 0},
  split_rngs={'params': False, 'other': True},
)

x = jnp.ones((3, 2))

params = BatchModel().init(jax.random.key(0), x)
print(params)
# {..., 'other_collection': {'kernel': 
Array([[ 0.0761425, -1.6157459],
            [ 0.0761425, -1.6157459],
            [ 0.0761425, -1.6157459]], dtype=float32)}}}
```

**Comment by krzysztofrusek** [comment]
OK , so I am closing this issue, thank you.
